common/transport/transport-api/src/main/java/org/thingsboard/server/common/transport/service/DefaultTransportService.java

```diff
@@ -1100,11 +1101,19 @@ protected void sendToDeviceActor(TransportProtos.SessionInfoProto sessionInfo, T
     }
 
     private void sendToCore(TenantId tenantId, EntityId entityId, ToCoreMsg msg, UUID routingKey, TransportServiceCallback<Void> callback) {
-        TopicPartitionInfo tpi = partitionService.resolve(ServiceType.TB_CORE, tenantId, entityId);
+        TopicPartitionInfo tpi;
+        try {
+            tpi = partitionService.resolve(ServiceType.TB_CORE, tenantId, entityId);
+        } catch (Exception e) {
+            log.trace("Failed to send message to core. Tenant with ID [{}], entityType [{}], entityId [{}], routingKey [{}], \nmsg [{}].\n Message delivery aborted.", tenantId, entityId.getEntityType(),  entityId.toString(), routingKey, msg, e);
```

Comment:
entityId.toString() or entityId.getEntitytype may fail with NPE. no need to call those getters .

Reply:
Ok